processor/groupbyattrsprocessor/factory.go

```diff
@@ -0,0 +1,115 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package groupbyattrsprocessor
+
+import (
+	"context"
+	"fmt"
+
+	"go.opencensus.io/stats/view"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configmodels"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/processor/processorhelper"
+	"go.uber.org/zap"
+)
+
+const (
+	// typeStr is the value of "type" for this processor in the configuration.
+	typeStr configmodels.Type = "groupbyattrs"
+)
+
+var (
+	errAtLeastOneAttributeNeeded = fmt.Errorf("option 'groupByKeys' must include at least one non-empty attribute name")
+	processorCapabilities        = component.ProcessorCapabilities{MutatesConsumedData: true}
+)
+
+// NewFactory returns a new factory for the Filter processor.
+func NewFactory() component.ProcessorFactory {
+	// TODO: as with other -contrib factories registering metrics, this is causing the error being ignored
+	_ = view.Register(MetricViews()...)
+
+	return processorhelper.NewFactory(
+		typeStr,
+		createDefaultConfig,
+		processorhelper.WithTraces(createTraceProcessor),
+		processorhelper.WithLogs(createLogsProcessor))
+}
+
+// createDefaultConfig creates the default configuration for the processor.
+func createDefaultConfig() configmodels.Processor {
+	return &Config{
+		ProcessorSettings: configmodels.ProcessorSettings{
+			TypeVal: typeStr,
+			NameVal: string(typeStr),
+		},
+		GroupByKeys: []string{},
+	}
+}
+
+func createGroupByAttrsProcessor(logger *zap.Logger, attributes []string) (*groupbyattrsprocessor, error) {
+	var nonEmptyAttributes []string
+	for _, str := range attributes {
+		if str != "" {
+			nonEmptyAttributes = append(nonEmptyAttributes, str)
+		}
+	}
+
+	if len(nonEmptyAttributes) == 0 {
+		return nil, errAtLeastOneAttributeNeeded
+	}
+
```

Comment:
Nit: maybe also validate that there are no duplicate keys and output warning in the log if there are.